terra-money/station-legacy
# Add TNS support

## Changes
- Introduce `helper` field on `Field` interface to show a subtle note about the resolved address.

![Screen_Shot_2564-11-08_at_17 21 56](https://user-images.githubusercontent.com/93900505/140726950-d066240d-ed01-4906-bf74-c10378afe21d.png)

- Add TNS name resolver to 3 pages:
  - Send token (`src/post/hooks/wallet/useSend.ts`)
  - Community Pool Spend Proposal (`src/post/hooks/gov/usePropose.ts`)
  - Sign-in with address (`src/hooks/auth/useSignInWithAddress.ts`)
- Since the validation function in `useForm` only supports synchronous function, TNS name resolution logic is done outside the `useForm` scope. Hence, `useTns` hook is introduced.
- Add TNS name error `This name has not yet been configured` and other languages translation.

## Test cases
- Fill in `blossom.ust`, it should resolve to an address `terra1...9hthwzp` (Like above image)
- Fill in `axfkcsnjr.ust`, it should show `This name has not yet been configured` validation error message.

![Screen_Shot_2564-11-08_at_17 30 24](https://user-images.githubusercontent.com/93900505/140726537-2be602c8-c9c8-44cd-af6b-61d35d89629a.png)


## Expected Behavior
- Send token
![station-wallet](https://user-images.githubusercontent.com/93900505/140726614-94ae6c4c-7580-4c91-bfe5-434e8d81c488.gif)

- Community Pool Spend Proposal
![station-gov](https://user-images.githubusercontent.com/93900505/140726686-57af0640-9c2f-4620-b5f6-945f75b125fc.gif)

- Sign-in with address
![station-connect](https://user-images.githubusercontent.com/93900505/140726435-bbfa5e41-6fdf-42d1-ac6f-8859c4e14a6c.gif)
